src/sortStudents.ts

```diff
@@ -1,16 +1,56 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function getAverageGrade({ grades }: Student): number {
+  const sum = grades.reduce((acc, grade) => acc + grade, 0);
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return sum / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const newStudents = [...students];
+
+  return newStudents.sort((aStudent, bStudent) => {
```

Comment:
In this case, you can avoid creating additional const

```suggestion
  return [...students].sort((aStudent, bStudent) => {
```